### PR TITLE
Process Agent Set Log Level

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?
Creates a server in the process agent that listens for requests to change the log_level. Also adds a client that can be accessed using the "config" subcommand. The port (default 6162) that the client and server uses can be configured using the process_config.cmd_port, or the DD_PROCESS_AGENT_CMD_PORT environment variable.

The default port 6162 is chosen because 6063 is taken by the X Window System

### Motivation
We wanted to mimic the behavior that other agents had, so we created initial support for changing runtime settings.

### Describe how to test your changes

Start the process agent. In another terminal, test the "config set log_level <log_level>", "config get log_level" and "config list-runtime" commands.

Try playing around with the port that the client and server listens on by using the process_config.cmd_port setting in config.yaml, and the DD_PROCESS_AGENT_CMD_PORT environment variable.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: releasenotes/notes/process-agent-log-level-config-8b6873639c21f411.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Initial support for changing running settings in the process agent is now included. For now the only configurable setting
+    is log_level. The port that is used for this server can be changed using either the DD_PROCESS_AGENT_CMD_PORT environment variable,
+    or the process_config.cmd_port setting in your config.yaml file.
```
Comment (xornivore) [inline releasenotes/notes/process-agent-log-level-config-8b6873639c21f411.yaml:13]: Maybe shorten this? Borrowing from an existing release note from agent:
```suggestion
    Adds the ability to change `log_level` in the process agent at runtime using ``process-agent config set log_level <log-level>``
```

**Review by xornivore — COMMENTED** [review]
Nice work!

File: cmd/process-agent/api/server.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+// StartServer starts the config server
+func StartServer() error {
+	// Set up routes
+	r := mux.NewRouter()
+	r.HandleFunc("/config", settingshttp.Server.GetValue)
+	r.HandleFunc("/config", settingshttp.Server.GetFull("")).Methods("GET")
+	r.HandleFunc("/config/list-runtime", settingshttp.Server.ListConfigurable).Methods("GET")
+	r.HandleFunc("/config/{setting}", settingshttp.Server.GetValue).Methods("GET")
+	r.HandleFunc("/config/{setting}", settingshttp.Server.SetValue).Methods("POST")
```
[the remark this answers is not in the document]
Comment (just-chillin) [reply cmd/process-agent/api/server.go]: Fixed :)

File: cmd/process-agent/api/server.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/settings"
+	settingshttp "github.com/DataDog/datadog-agent/pkg/config/settings/http"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/gorilla/mux"
+)
+
+// NOTE: Any settings you want to register should simply be added here
+var processRuntimeSettings = []settings.RuntimeSetting{
```
[the remark this answers is not in the document]
Comment (just-chillin) [reply cmd/process-agent/api/server.go]: Fixed :)

File: cmd/process-agent/api/server.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (just-chillin) [reply cmd/process-agent/api/server.go]: Fixed 👍🏻

File: cmd/process-agent/api/server.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (just-chillin) [reply cmd/process-agent/api/server.go]: Fixed 👍🏻

File: releasenotes/notes/process-agent-log-level-config-8b6873639c21f411.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
```
Comment (xornivore) [inline releasenotes/notes/process-agent-log-level-config-8b6873639c21f411.yaml:7]: Can drop this comment.

File: cmd/process-agent/api/server.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+import (
+	"fmt"
+	"net/http"
+
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	settingshttp "github.com/DataDog/datadog-agent/pkg/config/settings/http"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/gorilla/mux"
```
Comment (xornivore) [inline cmd/process-agent/api/server.go:11]: Move this import block to be above project imports (starting with ddconfig)

```suggestion
	"github.com/gorilla/mux"

	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
	settingshttp "github.com/DataDog/datadog-agent/pkg/config/settings/http"
	"github.com/DataDog/datadog-agent/pkg/util/log"
```

File: cmd/process-agent/api/server.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+func setupHandlers(r *mux.Router) {
+	r.HandleFunc("/config", settingshttp.Server.GetValue)
+	r.HandleFunc("/config", settingshttp.Server.GetFull("")).Methods("GET")
```
Comment (xornivore) [inline cmd/process-agent/api/server.go:16]: What happens when you run this one?

File: cmd/process-agent/api/server.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	return nil
+}
+
+// getIPCAddressPort returns a listening connection
+func getIPCAddressPort() (string, error) {
+	address, err := ddconfig.GetIPCAddress()
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%v:%d", address, ddconfig.Datadog.GetInt("process_config.cmd_port")), nil
```
Comment (wiyu) [inline cmd/process-agent/api/server.go:54]: Not consider using https://golang.org/pkg/net/#JoinHostPort

File: cmd/process-agent/api/server.go (hunk 5 above)
Comment (just-chillin) [reply cmd/process-agent/api/server.go:11]: Oh gotcha I thought it was supposed to be below

File: cmd/process-agent/api/server.go (hunk 7 above)
Comment (just-chillin) [reply cmd/process-agent/api/server.go:54]: Oh! I didn't know this was a thing! Thanks for pointing that out to me :)

File: cmd/process-agent/api/server.go (hunk 6 above)
Comment (just-chillin) [reply cmd/process-agent/api/server.go:16]: Oh!!!! This is actually a bug! Thank you so much for pointing that out to me

File: releasenotes/notes/process-agent-log-level-config-8b6873639c21f411.yaml (hunk 4 above)
Comment (just-chillin) [reply releasenotes/notes/process-agent-log-level-config-8b6873639c21f411.yaml:7]: Thanks, fixed! :)

File: pkg/config/config_template.yaml [hunk 8]
```diff
@@ -974,6 +974,11 @@ api_key:
   #
   # expvar_port: 6062
 
+  ## @param config_port - string - optional - default: 60621
+  ## Port for configuring runtime settings for the process Agent. 60621 is chosen because 6063 is used by the X Window System.
```
[the remark this answers is not in the document]
Comment (just-chillin) [reply pkg/config/config_template.yaml:978]: 6162 doesn't look like it's being used by anything important. I will go ahead and change it

**Review by cswatt — APPROVED** [review]
approved for docs

**Review by shang-wang — APPROVED** [review]
Very nice!

File: pkg/process/config/config.go [hunk 9]
```diff
@@ -371,6 +374,19 @@ func NewAgentConfig(loggerName config.LoggerName, yamlPath, netYamlPath string)
 	return cfg, nil
 }
 
+// RegisterRuntimeSettings registers settings to be added to the runtime config.
```
Comment (wiyu) [inline pkg/process/config/config.go:377]: nit: not sure if this is leftover refactor that didn't update the comment with the new function name.
```suggestion
// initRuntimeSettings registers settings to be added to the runtime config.
```

**Review by wiyu — APPROVED** [review]
💯

**Review by xornivore — APPROVED** [review]
🚀 Nice work, to be safe - could you please test it in Windows and using our helm chart on Kubernetes.

**Review by djmitche — APPROVED** [review]
`pkg/config/**` changes look good

**Pull request merged**